kzg_prover/src/utils/batched_kzg.rs

```diff
@@ -0,0 +1,104 @@
+use halo2_proofs::{
+    arithmetic::{best_fft, eval_polynomial, kate_division, Field},
+    halo2curves::{
+        bn256::{Bn256, Fr as Fp, G1Affine, G1},
+        group::{prime::PrimeCurveAffine, Curve, Group},
+    },
+    poly::{
+        commitment::{Blind, CommitmentScheme, ParamsProver},
+        kzg::commitment::ParamsKZG,
+        Coeff, EvaluationDomain, Polynomial,
+    },
+};
+
+pub fn commit_kzg(params: &ParamsKZG<Bn256>, poly: &Polynomial<Fp, Coeff>) -> G1 {
+    params.commit(poly, Blind::default())
+}
+
+/// Computes the polynomial h(X) for the batch KZG algorithm.
+pub fn compute_h(
+    params: &ParamsKZG<Bn256>,
+    f_poly: &Polynomial<Fp, Coeff>,
+    domain: &EvaluationDomain<Fp>,
+) -> Vec<G1Affine> {
+    let d = f_poly.len(); // Degree of the polynomial
+
+    println!("d: {}", d);
+    // Extract s_commitments from ParamsKZG and extend with neutral elements
+    let mut s_commitments_reversed = params
+        .get_g()
+        .iter()
+        .map(PrimeCurveAffine::to_curve)
+        .collect::<Vec<_>>();
+    s_commitments_reversed.reverse();
+
+    let mut s_fft: Vec<G1> = vec![G1::identity(); 2 * d];
+    s_fft[..d].copy_from_slice(&s_commitments_reversed[..d]);
+
+    // Prepare coefficients vector and zero-pad at the beginning
+    let mut c_fft = vec![Fp::zero(); 2 * d];
+    //Create a reversed copy of the polynomial
+    // let mut f_reversed = f_poly.to_vec();
+    // f_reversed.reverse();
+    c_fft[d..(2 * d)].copy_from_slice(&f_poly);
+
+    println!("c_fft and s_fft assigned");
+    // Assuming omega is a d-th root of unity, and nu is its square (2d-th root of unity)
```

Comment:
It seems the "2d-th root of unity" is not omega square.

Reply:
At first I didn't believe too, so I wanted to check it. I added
```
println!("nu^2d: {:?}", nu.pow_vartime(&[2 * d as u64]));
```
and the result is 
```
nu^2d: 0x0000000000000000000000000000000000000000000000000000000000000001
```

Reply:
I understand that $\nu^{2d} = 1$ for a "2d-th root of unity", but I am slightly confused about whether  $\nu^{d}$ should also equal 1. This seems to conflict with the property of $\nu$ being a primitive root, as described in section **1.3 on Discrete Fourier Transform**.

Reply:
Got you, it wasn't a primitive root indeed. Fixed it, but the results are still not matching.